### ISSUE TITLE
Fetching index of maximum values in tensors (after softmax) with device set as "xla" returns a constant value. 

### ISSUE THREAD

**Issue by 10-zin** [pytorch/xla#2169]
## 🐛 Bug

When a tensor is configured on "xla" device and a `softmax()` function is used (as shown in the example below), standard functions - `idx = torch.argmax()` or `_, idx = torch.max()` return a constant index value. In contrast, if the device is not "xla" then the functions return correct index values. 

## To Reproduce

```
import torch
import torch.nn.functional as F
normal_tensor = torch.tensor([[-0.4688,  0.2500],
        [ 1.5781, -1.3281],
        [-0.8203,  0.4863],
        [-2.4062,  2.0469]])
xla_tensor = torch.tensor([[-0.4688,  0.2500],
        [ 1.5781, -1.3281],
        [-0.8203,  0.4863],
        [-2.4062,  2.0469]], device='xla:1')

print(torch.argmax(F.softmax(xla_tensor, dim = 1), dim = 1))

print(torch.argmax(F.softmax(normal_tensor, dim = 1), dim = 1))
print(torch.argmax(xla_tensor, dim = 1))
print(torch.argmax(normal_tensor, dim = 1))
```

## Expected behavior
respective outputs

```
tensor([-9223372036854775808, -9223372036854775808, -9223372036854775808,
        -9223372036854775808], device='xla:1')

tensor([1, 0, 1, 1])
tensor([1, 0, 1, 1], device='xla:1')
tensor([1, 0, 1, 1])

```

## Environment

 - torch_xla version:  Nightly version.

**Comment by dlibenzi** [comment]
This works on XLA_CPU:
```Python
import torch
import torch.nn.functional as F
import torch_xla
import torch_xla.core.xla_model as xm

device = xm.xla_device()

normal_tensor = torch.tensor([[-0.4688, 0.2500], [1.5781, -1.3281],
                              [-0.8203, 0.4863], [-2.4062, 2.0469]])
xla_tensor = torch.tensor([[-0.4688, 0.2500], [1.5781, -1.3281],
                           [-0.8203, 0.4863], [-2.4062, 2.0469]],
                          device=device)

print(torch.argmax(F.softmax(xla_tensor, dim=1), dim=1))

print(torch.argmax(F.softmax(normal_tensor, dim=1), dim=1))
print(torch.argmax(xla_tensor, dim=1))
print(torch.argmax(normal_tensor, dim=1))
```

Let me try TPU ...

**Comment by dlibenzi** [comment]
Works on TPU as well:
```
tensor([1, 0, 1, 1], device='xla:1')
tensor([1, 0, 1, 1])
tensor([1, 0, 1, 1], device='xla:1')
tensor([1, 0, 1, 1])
```

**Comment by 10-zin** [comment]
Thanks, I checked this again in a new slate and works fine. Probably had some configuration and model specific problems before.
